### ISSUE TITLE
Fatal error when making a backup

### ISSUE THREAD

**Issue by Shelob9** [watchfulli/XCloner-Wordpress#192]
In a clean site (WP Version 5.8.3), with the current version installed through WordPress (xcloner 4.2.163) I am getting a fatal error when creating a backup.

## Steps To Reproduce

- Go to xCloner settings.
- Choose "Generate Backup" from button menu
- Use all default options
- Near end of update process, see error notification:

![Screenshot 2022-01-21 142532](https://user-images.githubusercontent.com/1994311/150588258-8eeb6d9b-eba9-472f-84d9-384090014aff.jpg)

I logged this error:

`Uncaught Error: Call to undefined method watchfulli\XClonerCore\Xcloner_File_System::directory() in plugins/xcloner/vendor/watchfulli/xcloner-core/src/Xcloner_Api.php:546`

**Comment by ovidiul** [comment]
👋  @Shelob9 

It might be worth changing that failing [line](https://github.com/watchfulli/xcloner-core/blob/0d4a20e8713e1a7052568f03418922e052dd885b/src/Xcloner_Api.php#L546) to:

```
$files = $this->xcloner_file_system->list_directory($folder);
```

**Comment by Shelob9** [comment]
I made the change to xcloner-core recommended by @ovidiul in a branch, https://github.com/watchfulli/xcloner-core/commit/1ecfd0ddde489360db1480414b25b719a557d6a0 Then I set the plugin to install that branch. I then went to my local dev site and made a new backup and it worked as expected. I am going to merge https://github.com/watchfulli/xcloner-core/pull/5 and then we can go back through testing the update.

Thanks so much @ovidiul for the fix! I'm still getting my head around how this all works.
